Week-3/4_Voting/OneVoteRound.sol

```diff
@@ -0,0 +1,92 @@
+
+contract oneVotingRound {
+    address admin;
+    bool votingInProgress;
+    bool votingStopped;
+    uint128 totalRegisteredCandidates;
+
+    address leadingCandidate;
+    uint256 leadingCandidateVotes;
+
+    struct Candidate {
+        address addr;
+        uint256 votes;
+    }
+
+
+    mapping(address => Candidate) CandidatesRecord; // to check if candidate is registered
+    mapping(address => bool) votersList; // to check if voters have voted or not
+
+    event DeclareWinner(address winner, uint256 votes);
+
+    modifier checkVotingStarted {
+        // Not allowing user to register if voting is started
+        require(!votingInProgress, "You are late. Voting already started.");
+        _;
+    }
+
+    modifier checkVotingStopped(string memory _msg) {
+        // Not allowing user to register if voting has stopped
+        require(!votingStopped, _msg);
+        _;
+    }
+
+    modifier isAdmin(string memory _msg) {
+        require(admin == msg.sender, _msg);
+        _;
+    }
+
+    // adding contract deployer as admin
+    constructor() {
+        admin = msg.sender;
+    }
+
+    function registerCandidate() external checkVotingStarted checkVotingStopped("You are late. Voting is completed.") {
+        // checking if candidate is registered or not
+        require(CandidatesRecord[msg.sender].addr == address(0), "You are already registered.");
+
+        Candidate memory newCandidate;
+        newCandidate.addr = msg.sender;
+
+        CandidatesRecord[msg.sender] = newCandidate;
+        totalRegisteredCandidates += 1;
+    }
+
+    function startVoting() external isAdmin("You are not authorised to start voting") checkVotingStarted checkVotingStopped("Voting is completed.") {
+        // checking if min 1 candidate is registered
+        require(totalRegisteredCandidates > 1, "No candidates registered");
+        votingInProgress = true;
+    }
+
+    function stopVoting() external isAdmin("You are not authorised to stop voting") checkVotingStopped("Voting already stopped") {
+        // checking if voting has started or not
+        require(votingInProgress, "Voting not yet started");
+        votingInProgress = false;
+        votingStopped = true;
+
+        emit DeclareWinner(leadingCandidate, leadingCandidateVotes);
+    }
+
+    function vote(address _voteTo) external checkVotingStopped("Voting has been stopped") {
+        // checking if voting has started
+        require(votingInProgress, "Voting not yet started");
+        // checking if voting sign is present
+        require(_voteTo != address(0), "Invalid candidate voting sign");
+        // checking if already voted
+        require(!votersList[msg.sender], "You have already voted");
+
+        // setting true if voted
+        votersList[msg.sender] = true;
+
+        // vote
+        uint _totalVotes = CandidatesRecord[_voteTo].votes + 1;
+        CandidatesRecord[_voteTo].votes = _totalVotes;
+
+        if(_totalVotes > leadingCandidateVotes){
```

Comment:
Directly increment CandidatesRecord[_voteTo].votes and check on line 87. We do not need to create new variable `_totalVotes` for the same

Reply:
done